ResumableUpload.CS.Sample/ResumeableUploadSample.cs

```diff
@@ -0,0 +1,241 @@
+        private static bool IsResumeable()
+        {
+            return ((Globals.ResumeRetries < MAX_RESUME_RETRIES) && (!objCancellationToken.IsCancellationRequested));
+        }
+        /// <summary>
+        /// The event handler called by the .NET Client Library to indicate the current status of the upload.
+        /// </summary>
+        /// <param name="objUploadStatus"></param>
+        static void videosInsertRequest_ProgressChanged(Google.Apis.Upload.IUploadProgress objUploadStatus)
+        {
+            switch (objUploadStatus.Status)
+            {
+                case UploadStatus.Completed:
+                    Globals.ResumeRetries = Int32.MaxValue; // Ensure that Do..While loop terminates
+                    Console.WriteLine("Status: Completed");
+                    break;
+                case UploadStatus.Starting:
+                    Console.WriteLine("Status: Starting");
+                    break;
+                case UploadStatus.NotStarted:
+                    Console.WriteLine("Status: Not Started");
+                    break;
+                case UploadStatus.Uploading:
+                    float sngPercent = 0;
+                    Globals.ResumeRetries = 0; // Successfully transmitted something so reset resume retries counter
+                    if (Globals.FileLength > 0)
+                        {
+                        sngPercent = (float)((100.0 * objUploadStatus.BytesSent) / Globals.FileLength);
+                        }
+                    //
+                    // Display only whole percents
+                    //
+                    int p = (int)sngPercent;
+                    if (p > Globals.ProgressPercent)
+                        {
+                        string strMessage =String.Format( "Status: Uploading {0:##0}% ({1:###,###,###,###,###,###,###,###})", p, objUploadStatus.BytesSent);
+                        Console.WriteLine(strMessage);
+                        Globals.ProgressPercent = (int)sngPercent;
+                        }
+                    break;
+                case UploadStatus.Failed:
+                    Console.WriteLine(string.Format("Upload Failed: {0}", objUploadStatus.Exception.Message));
+                    break;
+            }
+        }
+        /// <summary>
+        /// The event handler called by the .NET Client Library to provide the Video object containing the Id of the uploaded video.
+        /// </summary>
+        /// <param name="objVideo"></param>
+        static void videosInsertRequest_ResponseReceived(Video objVideo)
+        {
+            Console.WriteLine(string.Format("Video ID={0} uploaded.", objVideo.Id));
+        }
+
+    }
+    /// <summary>
+    /// Holds global variables required to computer percent complete and count retries.
+    /// </summary>
+    public static class Globals
+    {
+        // Globally Addressable Variables
+        public static long FileLength { get; set; }   // Used to compute percent complete
```

Comment:
If you put the comments above the properties, they will show up in intelli-sense. e.g.

    // <summary>Used to compute percent complete</summary>
    public static long FileLength {get; set; }